src/System.Globalization/tests/CultureInfo/CultureInfoCalendar.cs

```diff
@@ -2,19 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using Xunit;
 
 namespace System.Globalization.Tests
 {
     public class CultureInfoCalendar
     {
+        public static IEnumerable<object[]> Calendar_TestData()
+        {
+            yield return new object[] { CultureInfo.InvariantCulture, typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("en-US"), typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("th-TH"), typeof(ThaiBuddhistCalendar) };
+            yield return new object[] { new CultureInfo("ja-JP"), typeof(GregorianCalendar) };
+            yield return new object[] { new CultureInfo("ar-SA"), typeof(UmAlQuraCalendar) };
+        }
+
         [Theory]
-        [InlineData("en-US", typeof(GregorianCalendar))]
-        [InlineData("th-TH", typeof(ThaiBuddhistCalendar))]
-        public void Calendar(string locale, Type expected)
+        [MemberData(nameof(Calendar_TestData))]
+        public void Calendar(CultureInfo cultureInfo, Type expected)
         {
-            CultureInfo cultureInfo = new CultureInfo(locale);
             Assert.IsType(expected, cultureInfo.Calendar);
+            Assert.Same(cultureInfo.Calendar, cultureInfo.Calendar);
+        }
+
+        public static IEnumerable<object[]> OptionalCalendars_TestData()
+        {
+            yield return new object[] { CultureInfo.InvariantCulture, new Type[] { typeof(GregorianCalendar) } };
+            yield return new object[] { new CultureInfo("en-US"), new Type[] { typeof(GregorianCalendar), typeof(GregorianCalendar) } };
+            yield return new object[] { new CultureInfo("th-TH"), new Type[] { typeof(ThaiBuddhistCalendar), typeof(GregorianCalendar) } };
+            yield return new object[] { new CultureInfo("ja-JP"), new Type[] { typeof(GregorianCalendar), typeof(JapaneseCalendar), typeof(GregorianCalendar) } };
+            yield return new object[] { new CultureInfo("ar-SA"), new Type[] { typeof(UmAlQuraCalendar), typeof(HijriCalendar), typeof(GregorianCalendar), typeof(GregorianCalendar), typeof(GregorianCalendar), typeof(GregorianCalendar), typeof(GregorianCalendar) } };
+        }
+
+        [Theory]
+        [MemberData(nameof(OptionalCalendars_TestData))]
+        public void OptionalCalendars(CultureInfo cultureInfo, Type[] expected)
+        {
+            Calendar[] optionalCalendars = cultureInfo.OptionalCalendars;
+
+            Assert.Equal(expected.Length, optionalCalendars.Length);
+            for (int i = 0; i < optionalCalendars.Length; i++)
+            {
+                Assert.IsType(expected[i], optionalCalendars[i]);
```

Comment:
this is can easily break too. you are considering the order of the calendars will not change which is not true. actually changing the OS settings, you can get different order as we store first calendar from the settings. please try not to assume the order here.